cropduster/utils/image.py

```diff
@@ -145,9 +149,16 @@ def process_image(im, save_filename=None, callback=lambda i: i, nq=0, save_param
             save_params.setdefault('quality', get_jpeg_quality(new_images[0].size[0], new_images[0].size[1]))
         if im.format in ('JPEG', 'PNG') and JPEG_SAVE_ICC_SUPPORTED:
             save_params.setdefault('icc_profile', im.info.get('icc_profile'))
-        new_images[0].save(save_filename, **save_params)
-
-        return PIL.Image.open(save_filename)
+        img = new_images[0]
+        with default_storage.open(save_filename, 'w') as f:
+            buffer = BytesIO()
+            img.save(buffer, format=im.format)
```

Comment:
Looks like PIL's Image `save` method takes any file-like object, so we don't need to use a BytesIO buffer intermediary. It's possible to do this as:

```python
f = default_storage.open(save_filename, 'w')
img.save(f, format=im.format, **save_params)
f.seek(0)
pil_image = PIL.Image.open(f)
pil_image.filename = save_filename
return pil_image
```

As I was writing that out, I realized that `**save_params` was missing from the `save` call. Also note that I didn't use the context manager: it seems that there are limits to what you can do with a PIL Image that has had its filehandler closed. (You can't, for instance, re-save it—unless you first call `im.fp.open()`, in which case why use the context manager)

Reply:
Ah, that's great. Came to a similar conclusion on the context manager reviewing the test failures, but interestingly this seems to work in the admin, nonetheless.